### PR TITLE
Update BCFTOOLS/ISEC: additional arguments

### LINKED ISSUE #9764: update module: BCFTOOLS/ISEC
Add arguments based on the [documentation](https://samtools.github.io/bcftools/bcftools.html#isec)

```
1. file-list
2. regions-file
3. targets-file
```

### Is there an existing module for this?

- [x] I have searched for the existing module

### Is there an open PR for this?

- [x] I have searched for existing PRs

### Is there an open issue for this?

- [x] I have searched for existing issues

### Are you going to work on this?

- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by georgiakes**
## PR checklist

Closes #9764 

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Broadcast software version numbers to `topic: versions` - [See version_topics](https://nf-co.re/blog/2025/version_topics)
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/bcftools/isec/tests/main.nf.test [hunk 1]
```diff
@@ -63,7 +66,154 @@ nextflow_process {
...
+                    path("${process.out.results[0][1]}").list().findAll {
+                        it.getFileName().toString() != "0000.vcf.gz.tbi" && it.getFileName().toString() != "0001.vcf.gz.tbi"
+                    }
+                    ).match()
+                }
+            )
+        }
+    }
+
+    test("sarscov2 - [[vcf1, vcf2], [tbi1, tbi2]] - regions - stub") {
```
Comment (itrujnara) [inline modules/nf-core/bcftools/isec/tests/main.nf.test:199]: I don't see a test for the `file_list` input. Can you please add one? You can generate the list on the fly with a `setup` block if needed.

**Comment by georgiakes** [comment]
Should I also update tests to fix linting? It fails because the output files are empty.

**Comment by georgiakes** [comment]
@nf-core-bot fix linting
